### ISSUE TITLE
[RELAY][OP] Relay Op Sprint (Part 2)

### ISSUE THREAD

**Issue by jroesch** [dmlc/tvm#2051]
This is follow-up work to #1799. Now that we have merged an initial version of the Relay evaluator and runtime system in #1954 it is possible to use Relay for end-to-end inference, and optimization.

In order to do so we need to add attributes to the existing operators so that they may be correctly lowered to TVM.

# General Steps of Porting

In order to enable lowering of an operator we need to add both a scheduling and compute primitive.

#2050 shows an example for all of the ops in `tensor.py`
# The List

# The List

## Level 1: Common Basic Ops
- [x] nn.dense
- [x] nn.relu
- [x]  tanh
- [x]  sigmoid
- [x]  exp
- [x]  log
- [x]  sqrt
- [x]  add
- [x]  subtract
- [x]  multiply
- [x]  divide
- [x]  mod
- [x]  nn.batch_flatten
- [x]  concatenate
- [x]  nn.softmax
- [x]  nn.log_softmax
- [x]  nn.batch_norm
- [x]  nn.dropout
- [x] expand_dims

## Level 2: Convolutions

- [x]  nn.conv2d
- [x]  nn.conv2d_transpose
- [x]  nn.max_pool2d
- [x]  nn.avg_pool2d
- [x]  nn.global_max_pool2d
- [x]  nn.global_avg_pool2d
- [ ]  nn.pad
- [ ]  nn.lrn

## Level 3: Additional Math And Transform Operators

- [x]  reshape
- [x]  copy
- [x]  negative
- [x]  floor
- [x]  ceil
- [x]  round
- [x]  trunc
- [x]  clip
- [x]  abs
- [x]  leaky_relu
- [x]  tranpose
- [x]  split
- [x]  squeeze
- [x]  take
- [ ]  full
- [x]  zeros
- [x]  ones
- [x]  transpose
- [x]  zeros_like
- [x]  ones_like

## Level 4: All broadcast and reduction functions that are not in previous level

- [x]  pow
- [x]  less
- [x]  greater
- [x]  less_than
- [x]  greater_than
- [x]  right_shift
- [x]  left_shift
- [x]  maximum
- [x]  minimum
- [x]  sum
- [x]  max
- [x]  prod
- [x]  argmax, argmin
- [x]  strided_slice
- [ ]  broadcast_to
- [x]  where

## Level 5: Vision Operators

- [x]  image.resize
- [ ]  vision.multibox_prior
- [ ]  vision.nms

**Comment by MarisaKirisame** [comment]
dense, softmax, log_softmax, concatenate, dropout

**Comment by masahi** [comment]
I noticed that the layout_transform op is missing in the above list ans also in #1799. This is important for cpu inference. Is anybody working on this? @yzhliu @kevinthesun

**Comment by kevinthesun** [comment]
@masahi layout_transform needs to be refactored, since the current implementation is in nnvm instead of topi. This is also important for graph tuner. @yzhliu Do we have any plan?

**Comment by yzhliu** [comment]
Yes I'm working on it.

**Comment by tqchen** [comment]
Thanks to everyone's effort, most of the ops are now supported. Let us open new threads to catch the remaining outliers when necessary
